app/models/spell_finder.rb

```diff
@@ -0,0 +1,26 @@
+class SpellFinder
+  attr_reader :class_name, :level, :page
+
+  def initialize(params)
+    @params = params
+    @class_name = params[:class_name]
+    @level = params[:level].try(:to_i)
+    @page = params[:page]
+  end
+
+  def self.perform(params)
+    new(params).perform
+  end
+
+  def perform
+    # TODO: This can and should be cleaned up to a single composable query but I can't think of how just yet
+    # it may involve abstracting out the json to a join table
```

Comment:
I'm not sure how it would be only one query.

A user should be able to filter by class or by class and level. And what about searching? I don't think that would that be done through a separate model.

Reply:
The general idea is that we can do a lot with a sql query. It might be that we can compose differernt parts of one query and then put the different pieces together to get the result we want. Search might end up being a different method.